src/main/java/appeng/items/tools/powered/ToolMatterCannon.java

```diff
@@ -124,17 +124,22 @@ public void addCheckedInformation( final ItemStack stack, final World world, fin
 				shots += cu.getInstalledUpgrades( Upgrades.SPEED );
 			}
 
-			final IMEInventory inv = AEApi.instance().registries().cell().getCellInventory( p.getHeldItem( hand ), null,
-					AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
+			final ICellInventoryHandler<IAEItemStack> inv = AEApi.instance()
+					.registries()
+					.cell()
+					.getCellInventory( p.getHeldItem( hand ), null,
+							AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
 			if( inv != null )
 			{
-				final IItemList itemList = inv.getAvailableItems( AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ).createList() );
-				IAEStack aeAmmo = itemList.getFirstItem();
-				if( aeAmmo instanceof IAEItemStack )
+				final IItemList<IAEItemStack> itemList = inv
+						.getAvailableItems( AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ).createList() );
+				IAEItemStack req = itemList.getFirstItem();
+				if( req instanceof IAEItemStack )
 				{
-					shots = Math.min( shots, (int) aeAmmo.getStackSize() );
+					shots = Math.min( shots, (int) req.getStackSize() );
 					for( int sh = 0; sh < shots; sh++ )
 					{
+						IAEItemStack aeAmmo = req.copy();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "aeAmmo" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)